scout/utils/ensembl_biomart_clients.py

```diff
@@ -0,0 +1,49 @@
+import logging
+from typing import Callable, Dict, Iterator
+
+import requests
+from schug.load.biomart import EnsemblBiomartClient
+from schug.load.ensembl import fetch_ensembl_exons, fetch_ensembl_genes, fetch_ensembl_transcripts
+from schug.models.common import Build as SchugBuild
+
+LOG = logging.getLogger(__name__)
+
+BUILDS: Dict[str, str] = {"37": "GRCh37", "38": "GRCh38"}
+
+ENSEMBL_RESOURCE_CLIENT: Dict[str, Callable] = {
+    "genes": fetch_ensembl_genes,
+    "transcripts": fetch_ensembl_transcripts,
+    "exons": fetch_ensembl_exons,
+}
+
+
+class EnsemblBiomartHandler:
+    """A class that handles Ensembl genes, transcripts and exons downloads via schug."""
+
+    def __init__(self, build: str = "37"):
+        self.build: str = BUILDS[build]
+
+    def biomart_get(self, url: str) -> Iterator:
+        """Sends a request to Ensembl Biomart and returns the resource lines."""
+        response: requests.models.responses = requests.get(url, stream=True)
+        return response.iter_lines(decode_unicode=True)
+
+    def stream_resource(self, interval_type: str) -> Iterator[str]:
+        """Fetches genes, transcripts or exons from a remote Ensembl biomart in the right genome build and saves them to file."""
+
+        def yield_resource_lines(iterable) -> str:
```

Comment:
Hm, isn't this just 
`iterable.pop(); return iterable` - or `iterable.pop(-1)` if it was on the other end?

Reply:
Didn't know it existed! I'll give it a try!

Reply:
Ah, ok; I've learned something! All iterables are not alike, and while many of them have a pop() implementation, it is presumably not in the base class. I suppose one could either just instantiate as list or other pop-able iterator, or use "generator comprehension", but all good - this works, I was mostly curios!

Reply:
Absolutely - this is not wrong! One can do it differently, but pop that I suggested was not available!